boa/dealer.py

```diff
@@ -0,0 +1,90 @@
+import boa
+from boa.util.abi import Address
+
+SLOAD_OPCODE = 0x54
+
+
+class SloadTracer:
+    def __init__(self, super_fn):
+        self.super_fn = super_fn
+        self.trace = []
+
+    def __call__(self, computation):
+        # TODO find a way to peak instead of pop/repush
+        slot = computation.stack_pop1_int()
+        computation.stack_push_int(slot)
+        self.super_fn(computation)
+        self.trace.append(slot)
+
+
+def find_storage_slot(contract, function_selector: str, *args):
+    # we use instantiate a tracker to track storage slots accessed by the SLOAD opcode
+    sload_tracer = SloadTracer(
+        boa.env.evm.vm.state.computation_class.opcodes[SLOAD_OPCODE]
+    )
+    boa.patch_opcode(SLOAD_OPCODE, sload_tracer)
+
+    try:
+        result = getattr(contract, function_selector)(*args)
+    except AttributeError:
+        raise ValueError(f"Function {function_selector} not found in {contract}")
+
+    # we iterate over all the storage slots accessed by the SLOAD opcode
+    # until we find the one that contains the result of the function call
+    target_slot = None
+    for slot in sload_tracer.trace:
+        slot_value = boa.env.get_storage(contract.address, slot)
+
+        if slot_value == result:
+            # sanity check in a sandboxed environment to avoid false positives
+            with boa.env.anchor():
+                boa.env.set_storage(contract.address, slot, 123456789)
+                if getattr(contract, function_selector)(*args) != 123456789:
+                    continue
+
+            # if the slot contains the result, and it's not a false positive
+            # we found the target slot
+            target_slot = slot
+            break
```

Comment:
```suggestion
                if getattr(contract, function_selector)(*args) == 123456789:
                    target_slot = slot
                    break
```

Reply:
or you could just `return` straight out of here, and simplify the branch on line 52